tests/app/authentication/test_jwt.py

```diff
@@ -0,0 +1,285 @@
+import unittest
+from tests.app.authentication import SR_PRIVATE_PEM
+from app.authentication.jwt_decoder import Decoder
+from app.authentication.invalid_token_exception import InvalidTokenException
+import base64
+import os
+
+# public key from jwt.io
+
+PUBLIC_KEY = """-----BEGIN PUBLIC KEY-----
+MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDdlatRjRjogo3Wojg
+GHFHYLugdUWAY9iR3fy4arWNA1KoS8kVw33cJibXr8bvwUAUparCwlv
+dbH6dvEOfou0/gCFQsHUfQrSDv+MuSUMAe8jzKE4qW+jK+xQU9a03GU
+nKHkkle+Q0pX/g6jXZ7r1/xAK5Do2kQ+X5xK9cipRgEKwIDAQAB
+-----END PUBLIC KEY-----"""
+
+# jwt.io public key signed
+
+jwtio_header = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IkVEQ1JSTSJ9"
+jwtio_payload = "eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlhdCI6IjE0NTQ5MzU3NjciLCJleHAi" \
+                "OiIyMDc1Mjk3MTQ4In0"
+jwtio_signature = "1PVw5ku4uzif8iaXpbEhsSkbnug5mXMYMvBUUpGQyJ73mqtl7oApntEyC-z9jSunqAA0JytOVnV1s39PbCE1tfFtZIcyEddC4d" \
+                  "M7oCHAXEjZukUnhX5XQDC323zGQbbOOgoaldHYJU8I_pdbILEtK7QFPKB4mkvEH2YhAIQya3s"
+
+jwtio_signed = jwtio_header + "." + jwtio_payload + "." + jwtio_signature
+
+
+class JWTTest(unittest.TestCase):
+
+    def test_jwt_io(self):
+        decoder = Decoder(PUBLIC_KEY, SR_PRIVATE_PEM, "digitaleq")
+        token = decoder.decode_signed_jwt_token(jwtio_signed)
+        self.assertEquals("John Doe", token.get("name"))
+
+    def test_does_not_contain_two_instances_of_full_stop(self):
+        jwe = jwtio_signed.replace('.', '', 1)
+        decoder = Decoder(PUBLIC_KEY, SR_PRIVATE_PEM, "digitaleq")
+        with self.assertRaises(InvalidTokenException) as ite:
+            decoder.decode_signed_jwt_token(jwe)
+        self.assertIn("Error", ite.exception.value)
+
+    def test_jwt_does_not_contain_header(self):
+        token_without_header = "e30." + jwtio_payload + "." + jwtio_signature
+
+        decoder = Decoder(PUBLIC_KEY, SR_PRIVATE_PEM, "digitaleq")
+        with self.assertRaises(InvalidTokenException) as ite:
+            decoder.decode_signed_jwt_token(token_without_header)
+        self.assertIn("Missing Headers", ite.exception.value)
+
+    def test_jwt_does_not_contain_payload(self):
+        token_without_payload = jwtio_header + ".e30." + jwtio_signature
```

Comment:
Is there a test that has an empty payload e.g. jwtio_header + ".." + jwtio_signature